amz_review_scraper/soup_searcher.py

```diff
@@ -1,11 +1,20 @@
 
+
```

Comment:
This is so mich nicer being isolated like this!

Something still feels wrong to be about that try except block for find the value. Not sure the best way forward.

But one random refactor you can look out for, is when you are iterating over a value then performing a bunch of stuff, to separate out the work, from the iterator. So just extract out a method.

Here's an example of a refactor, that was still passing your tests, so I am assuming works (yeah tests!)

```
def find_attribute(soup, key, html_tag, attrs):
    for tag in soup.findAll(html_tag, attrs=attrs):
        attribute = _find_attribute(tag, key, attrs)
        if attribute is not None:
            return attribute

    return "None"


def _find_attribute(tag, key, attrs):
    # For the Title, or Reviews Count inner function
    if attrs == {"id": "productTitle"} or attrs == {"id": "acrCustomerReviewText"}:
        return tag.text.strip()
    # For Reviews Count Outer function
    elif attrs == {"id": "averageCustomerReviews_feature_div"}:
        return find_attribute(
            tag, None, "span", attrs={"id": "acrCustomerReviewText"}
        )
    else:
        if tag.get(key) is not None:
            return tag[key]
```

This also makes me wonder why return "None" the string instead of just None. Passing back actual None makes more sense, so then its easier to interact with more parts of the Python Ecosystem that depend on checking is something is `None`.

Also then you can just eliminate that last line, since None is return when there is no return statement by default.

```
def find_attribute(soup, key, html_tag, attrs):
    for tag in soup.findAll(html_tag, attrs=attrs):
        attribute = _find_attribute(tag, key, attrs)
        if attribute is not None:
            return attribute
```

also that actually passes the tests currently 🤷‍♂️, so maybe you don't even need other changes.

Also you don't use the `product_container` anywhere!

Reply:
i guess i was thinking that "None" would be storable in the database and None would not, but maybe Im wrong.